samples/highcharts/demo/flame/demo.js

```diff
@@ -0,0 +1,1205 @@
+    }, {
+        "name": "~",
+        "id": "2",
+        "value": 397316,
+        "color": colors[0],
+        "x": -3,
+        "low": 0,
+        "high": 397316
+    }, {
+        "name": "(root)",
+        "id": "1",
+        "value": 397316,
+        "color": colors[2],
+        "x": -2,
+        "low": 0,
+        "high": 397316
+    }];
+    
+(function(H){
+    H.seriesType('flame', 'columnrange', {
+        cursor: 'pointer',
+        dataLabels: {
+            enabled: true,
+            format: '{point.name}',
+            inside: true,
+            align: 'center',
+            crop: true,
+            overflow: 'none',
+            color: 'black',
+            style: {
+                textOutline: 'none',
+                fontWeight: 'normal'
+            }
+        },
+        point: {
+            events: {
+                click: function() {
+                    var point = this,
+                        chart = point.series.chart,
+                        series = point.series,
+                        xAxis = series.xAxis,
+                        yAxis = series.yAxis;
+
+                    xAxis.setExtremes(xAxis.min, point.x, false);
+                    yAxis.setExtremes(point.low, point.high, false);
+                    
+                    chart.showResetZoom();
+                    chart.redraw();
+                }
+            }
+        },
+        pointPadding: 0,
+        groupPadding: 0,
+    }, {
+            drawDataLabels: H.seriesTypes.line.prototype.drawDataLabels
+    });
+}(Highcharts))
+
+Highcharts.chart('container', {
+    chart: {
+        inverted: true,
+        marginTop: 100
+    },
+    title: {
+        text: 'Flame chart (layout: flame)',
+        y: 70
+    },
+    subtitle: {
+        text: 'Highcharts chart rendering process',
+        y: 85
+    },
+    legend: {
+            enabled: false
+    },
+    xAxis: [{
+        visible: false
+    }, {
+        visible: false,
+        startOnTick: false,
+        endOnTick: false
+    }],
+    yAxis: [{
+            visible: false
+    }, {
+        visible: false,
+        min: 0,
+        maxPadding: 0,
+        startOnTick: false,
+        endOnTick: false
+    }],
+    series: [{
+        type: 'flame',
+        data: data,
+        yAxis: 1,
+        xAxis: 1
+    }, {
+            visible: false,
+        size: '100%',
+        type: 'sunburst',
+        data: sunburstData,
+        allowDrillToNode: true,
+        cursor: 'pointer',
+        levels: [{
+            level: 1,
+            levelIsConstant: false,
+            dataLabels: {
+                enabled: false
+            }
+        }],
+        dataLabels: {
+            textPath: {
+                attributes: {
+                    dy: 5,
+                },
+                enabled: true
+            }
+        }
+    }],
+    tooltip: {
+        headerFormat: "",
+        pointFormat: 'selfSize of <b>{point.name}</b> is <b>{point.value}</b>'
+    }
+}, function(chart) {
```

Comment:
Add this functionality to the `load` event instead, so that we get a clean configuration of the chart instance. Then the event handler would apply to all charts in the page, but it should run a check before adding the buttons. For example check that the chart has a `flame` series, or that it has _only_ flame series. This event should be added within the same IIFE that adds the series type, so that it is an easily identifiable entity in the demo.

Reply:
On second thought, I actually think this kind of toolbar should be implemented in HTML. The only reason for adding it in the chart SVG is to preserve it in the exported chart, but dynamic buttons don't really make sense in the exported chart, you can't click them anyway...

Reply:
Changed to HTML buttons format and style based on [this Gantt demo with buttons](https://www.highcharts.com/gantt/demo/interactive-gantt).